sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -1,48 +1,112 @@
-# Temporarily assuming only dna components to be dealt with in genbank files
-COMP_TYPES = [sbol3.SBO_DNA]                
-# Temporarilty assuming components to only have the engineered_region role
-COMP_ROLES = [sbol3.SO_ENGINEERED_REGION]   
-GENBANK_FILE_1 = os.path.join(os.path.abspath(os.path.curdir),
-                    "test/test_files", SAMPLE_GB_FILE_1)
-SBOL3_FILE_1 = os.path.join(os.path.abspath(os.path.curdir),
-                    "test/test_files", SAMPLE_SBOL3_FILE_1)
-
-def convert_genbank_to_sbol3(gb_file: str, sbol3_file: str, 
-                    namespace: str = "https://testing.sbol3.genbank/", write: bool = False):
-    # create sbol3 document, and record parser handler for gb file
-    sbol3.set_namespace(namespace)
-    doc = sbol3.Document()
-    record = GENBANK_PARSER.parse(open(gb_file))
-    # NOTE: Currently we assume only linear or circular topology is possible
-    COMP_TYPES.append(sbol3.SO_LINEAR if record.topology == "linear" else sbol3.SO_CIRCULAR)
-    comp = sbol3.Component(identity=record.accession[0],
-                    types=COMP_TYPES,
-                    roles=COMP_ROLES,
-                    description=record.definition)
-    doc.add(comp)
-    # NOTE: Currently we use a single method of encoding (IUPAC)
-    seq = sbol3.Sequence(identity=record.accession[0] + "_sequence",
-                    elements=record.sequence.lower(),
-                    encoding=sbol3.IUPAC_DNA_ENCODING)
-    doc.add(seq)
-    comp.sequences = [seq]
-    if write: 
-        doc.write(fpath=sbol3_file, file_format=sbol3.SORTED_NTRIPLES)
-    return doc
+TEST_NAMESPACE = "https://test.sbol3.genbank/"
+# TODO: Temporarily assuming only dna components to be dealt with in genbank files
+COMP_TYPES = [sbol3.SBO_DNA]
+# TODO: Temporarily assuming components to only have the engineered_region role
+COMP_ROLES = [sbol3.SO_ENGINEERED_REGION]
+# TODO: Need to find a way to infer SequenceFeature Range/Cut Orientation from the genbank file
+SEQ_FEAT_RANGE_ORIENTATION = "https://identifiers.org/SO:0001030"  # inline orientation
+# TODO: Temporarily encoding sequnce objects in IUPUC mode only
+SEQUENCE_ENCODING = sbol3.IUPAC_DNA_ENCODING
+# Complete file paths to pass as paramteres for testing
+SAMPLE_GENBANK_FILE_1 = os.path.join(os.path.abspath(os.path.curdir), 
+                                "test/test_files", "BBa_J23101.gb")
+SAMPLE_SBOL3_FILE_1 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "BBa_J23101_from_genbank_to_sbol3_direct.nt")
+SAMPLE_GENBANK_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports.gb")
+SAMPLE_SBOL3_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports_from_genbank_to_sbol3_direct.nt")
+GB2SO_MAPPINGS = {}
+SO2GB_MAPPINGS = {}
+SO_SEQ_FEAT_ROLE_NS = "http://identifiers.org/so/"
```

Comment:
Is this the right namespace? pySBOL3 uses `https://identifiers.org/SO:`.

Reply:
Actually, I found the NS which I wrote in `test/test_files/iGEM_SBOL2_imports.nt` . But I see that both of our URLs do resolve (eg: https://identifiers.org/SO:0000167 and http://identifiers.org/so/SO:0000167). What should I finally keep here?

Reply:
See below, I have asked a question about the use of `sbol3.SO_NS`. I'm closing this conversation as resolved since you fixed the thing that concerned me here.